fhir-server/src/main/java/au/csiro/pathling/fhirpath/UntypedResourcePath.java

```diff
@@ -56,13 +57,15 @@ protected UntypedResourcePath(@Nonnull final String expression,
    * resource
    * @param valueColumn A {@link Column} within the dataset containing the values of the nodes
    * @param singular An indicator of whether this path represents a single-valued collection
+   * @param thisColumn collection values where this path originated from {@code $this}
    * @param typeColumn A {@link Column} within the dataset containing the resource type
    * @param possibleTypes A set of {@link ResourceType} objects that describe the different types
    * @return a shiny new UntypedResourcePath
    */
   public static UntypedResourcePath build(@Nonnull final String expression,
       @Nonnull final Dataset<Row> dataset, @Nonnull final Optional<Column> idColumn,
-      @Nonnull final Column valueColumn, final boolean singular, @Nonnull final Column typeColumn,
+      @Nonnull final Column valueColumn, final boolean singular,
+      @Nonnull final Optional<Column> thisColumn, @Nonnull final Column typeColumn,
       @Nonnull final Set<ResourceType> possibleTypes) {
 
     final String hash = Integer.toString(Math.abs(dataset.hashCode()), 36);
```

Comment:
`36` seems to be a magic number. Maybe `java.lang.Character.MAX_RADIX` instead?

Reply:
See c63665963d84106a6c678dcf884bb2e4b23a7501.